indexing-service/src/main/java/org/apache/druid/indexing/overlord/ZkWorker.java

```diff
@@ -63,6 +67,25 @@ public ZkWorker(Worker worker, PathChildrenCache statusCache, final ObjectMapper
     this.statusCache = statusCache;
     this.cacheConverter = (ChildData input) ->
         JacksonUtils.readValue(jsonMapper, input.getData(), TaskAnnouncement.class);
+    this.taskIdExtractor = createTaskIdExtractor(jsonMapper);
+  }
+
+  static java.util.function.Function<ChildData, String> createTaskIdExtractor(final ObjectMapper jsonMapper)
+  {
+    return (ChildData input) -> {
+      try (JsonParser parser = jsonMapper.getFactory().createParser(input.getData())) {
+        while (parser.nextToken() != JsonToken.END_OBJECT) {
```

Comment:
wouldn't this stop parsing on the end of the first nested object if the ordering of the fields is not guaranteed to have "id" come first?

Reply:
Likely, although this hasn't been an issue in my experience, it could become one;

I'll add tests and improve code to match.

Reply:
I don't think we offer any ordering guarantees today in the serialization, so this could break without warning